src/zenml/visualizers/base_step_visualizer.py

```diff
@@ -15,12 +15,19 @@
 from abc import abstractmethod
 from typing import Any
 
+from IPython import get_ipython  # type: ignore
+
 from zenml.logger import get_logger
 from zenml.post_execution import StepView
 from zenml.visualizers.base_visualizer import BaseVisualizer
 
 logger = get_logger(__name__)
 
+if get_ipython().__class__.__name__ == "TerminalInteractiveShell":
+    raise RuntimeError(
```

Comment:
Ok. I think I was confused because of the proposed solution in the ticket, which suggested targetting things in the `base_step_visualizer`. If we're just going to defer to the behaviour of normal Python, I reckon we'll have to do this in the actual individual visualization steps, if I'm not mistaken?

Reply:
yeah sorry for the confusion @strickvl -- when I wrote that ticket, the code was actually in base_step_visualizer. but since the writing of that ticket, apparently Michael refactored it into the previously non existing environment class. oops. I think the only place it now needs to be changed is in the environment class. we can sync orally in a min too